### DISCUSSION TITLE
Reading Label Confidence via API

### DISCUSSION THREAD

**Discussion by tilusnet** [photoprism/photoprism#4305] [Q&A] [answered]
Hello,

I've been skimming through the [Go API](https://pkg.go.dev/github.com/photoprism/photoprism/internal/api#pkg-functions) for some time now;
how would one get the confidence scores attached to a label in a photo?

I would also like to get the Source value, ie. "image", "manual".

Help is appreciated in advance!

![image](https://github.com/photoprism/photoprism/assets/705606/49301673-b1da-4269-944c-52fd385370e3)

**Comment by tilusnet** [comment]
Is this info impossible to get via API? Nobody knows here? 
I would be keen to know. Thanks.

**Reply by graciousgrey** [reply, marked as answer]
Hello :)

The [GetPhoto function](https://github.com/photoprism/photoprism/blob/61057a700a60/internal/api/photos.go#L45) returns, among other things, the labels that are assigned to a photo - including the source and the uncertainty. The confidence is 100 - the uncertainty.

**Comment by tilusnet** [comment]
Thank you @graciousgrey.
I tested with the API and I now see the nested 'Labels' structure with the Uncertainty field.

Can you please recommend an easy way for retrieving the API data schema?     
For instance, how can one tell what data is returned by the [GetPhoto function](https://github.com/photoprism/photoprism/blob/61057a700a60/internal/api/photos.go#L45) without actually testing the API call?

**Reply by lastzero** [reply]
The data fields returned are specified in the photo entity model: https://github.com/photoprism/photoprism/blob/develop/internal/entity/photo.go

**Comment by tilusnet** [comment]
Thank you @lastzero.

I failed to find that file myself - probably because I am not a Go coder. :)
I was usually looking in `iternal/form` only.
